### PR TITLE
DDP-5128 save invite in local storage

### PR DESCRIPTION
**Pull request by zyme**
## Context

DDP-5128 adds local storage persistence of `invitation_guid` to work around Auth0 limitations related to the breached password shield.

The change is remarkably simple, although it has its limitations: if we get an `invitation_guid` on the URL for the hosted login page, we save it to local storage and use it in subsequent logins if it's present.  If a different guid shows up on the URL in future calls, the URL-based value "wins".

## Checklist

- [ ] I have labeled the type of changes involved using the `C-*` labels.
- [ ] I have assessed potential risks and labeled using the `R-*` labels.
- [ ] I have considered error handling and alerts, and added `L-*` labels as needed.
- [ ] I have considered security and privacy, and added `I-*` labels as needed
- [ ] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.
- [ ] If applicable, I have discussed the analytics needs at both a platform and study level with Product and instrumented code accordingly.
- [ ] If applicable, my UI/UX changes have passed muster with Product/Design via an over-the-shoulder review, screenshots, etc.

_If unsure or need help with any of the above items, add the `help wanted` label. For items that starts with `If applicable`, if it is not applicable, check it off and add `n/a` in front._

## FUD Score

_Overall, how are you feeling about these changes?_

- [x ] :relaxed: All good, business as usual!

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [ ] They are user-visible in dev as a regular user journey and require no additional instructions.
- [ x] Getting dev into a state where this is user-visible requires some tech fiddling. I have documented these steps in the related ticket.
- [ ] Requires other features before it's human visible. I have documented the blocking issues in jira.
- [ ] I have no idea how to demo this. Please help me!

## Testing

- [ ] I have written automated positive tests
- [ ] I have written automated negative tests
- [ ] I have written zero automated tests but have poked around locally to verify proper functionality
- [ x] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [ ] These changes require no special release procedures--just code!
- [x ] Releasing these changes requires special handling and I have documented the special procedures in the release plan document

### CODE REVIEW

**Comment by MocanaAtBroad** [comment]
Question: Do auth0 rules execute at all  in the breached password scenario?
(Nevermind. You answered this question in the JIRA ticket).

File: study-builder/tenants/testboston/pages/login.html [hunk 1]
```diff
@@ -78,6 +78,20 @@
     // Decode utf8 characters properly
     var config = JSON.parse(decodeURIComponent(escape(window.atob('@@config@@'))));
     config.extraParams = config.extraParams || {};
+    var inviteKey = 'invitationId';
+
+    // save the invitation in local storage to minimize breached password loss-of-invitation
+    if (config.extraParams.invitation_id) {
+        localStorage.setItem(inviteKey,config.extraParams.invitation_id);
+        console.log('Saving invitation ' + config.extraParams.invitation_id);
```
Comment (MocanaAtBroad) [inline study-builder/tenants/testboston/pages/login.html:86]: Maybe just write the last 4 characters to console?

**Comment by MocanaAtBroad** [comment]
Is user info available in the breach password?

**Review by MocanaAtBroad — APPROVED** [review]

File: study-builder/tenants/testboston/pages/login.html (hunk 1 above)
Comment (zyme) [reply study-builder/tenants/testboston/pages/login.html:86]: Yes...but I'm not sure why?

Comment (MocanaAtBroad) [reply study-builder/tenants/testboston/pages/login.html:86]: Just paranoia.

**Review by fat23cat — APPROVED** [review]

**Pull request merged**